setup.py

```diff
@@ -170,6 +172,9 @@ def __exit__(self, type, value, traceback):
                   'hyperspy.external.astroML',
                   ],
         requires=install_req,
+        setup_requires=[
+            'setuptools'
```

Comment:
I don't understand these lines. How can `setuptools` be in `setup_requires`? Basically this field is only read if `setuptools` exists. So, this is either a no-op or a problem waiting to happen. So, why was this added then?